darwin/dataset/remote_dataset.py

```diff
@@ -379,33 +400,66 @@ def create_annotation_class(self, name: str, type: str):
             error_handlers=[name_taken, validation_error],
         )
 
-    def add_annotation_class(self, annotation_class):
+    def add_annotation_class(self, annotation_class: AnnotationClass) -> Union[Dict, None]:
+        """
+        Adds an annotation class to this dataset.
+
+        Parameters
+        ----------
+        annotation_class : AnnotationClass
+            The annotation class to add.
+       
+        Returns
+        -------
+        dict or None
+            Dictionary with the server response or None if the annotations class already exists.
+        """
         # Waiting for a better api for setting classes
         # in the meantime this will do
         all_classes = self.fetch_remote_classes(True)
-        annotation_class_type = (annotation_class.annotation_internal_type or annotation_class.annotation_type)
+        annotation_class_type = annotation_class.annotation_internal_type or annotation_class.annotation_type
         match = [
             cls
             for cls in all_classes
-            if cls["name"] == annotation_class.name
-            and annotation_class_type in cls["annotation_types"]
+            if cls["name"] == annotation_class.name and annotation_class_type in cls["annotation_types"]
         ]
         if not match:
             # We do not expect to reach here; as pervious logic divides annotation classes in imports
             # between `in team` and `new to platform`
-            raise ValueError(f"Annotation class name: `{annotation_class.name}`, type: `{annotation_class_type}`; does not exist in Team.")
+            raise ValueError(
+                f"Annotation class name: `{annotation_class.name}`, type: `{annotation_class_type}`; does not exist in Team."
+            )
 
         datasets = match[0]["datasets"]
         # check that we are not already part of the dataset
         for dataset in datasets:
             if dataset["id"] == self.dataset_id:
-                return
+                return None
         datasets.append({"id": self.dataset_id})
+        # we typecast to dictionary because we are not passing the raw=True parameter.
         return self.client.put(f"/annotation_classes/{match[0]['id']}", {"datasets": datasets, "id": match[0]["id"]})
 
-    def fetch_remote_classes(self, team_wide=False):
-        """Fetches all remote classes on the remote dataset"""
+    def fetch_remote_classes(self, team_wide=False) -> Optional[List]:
+        """
+        Fetches all remote classes on the remote dataset.
+
+        Parameters
+        ----------
+        team_wide : bool
+             If `True` will instead get the Annotation Classes from all Datasets that belong this
+             Dataset's team.
+       
```

Comment:
Annotation classes do not belong to datasets anymore; so we shouldn't use that phrasing in our docs else it could create the wrong mental model for clients.

If `True` will return all Annotation Classes that belong to the team. If `False` will only return Annotation Classes which have been added to the dataset.

Reply:
Fixed !